### PR TITLE
Implemented personal token authorization creation

### LINKED ISSUE #969: Create a new authorization
I'm needing to [create a new authorization](https://developer.github.com/v3/oauth_authorizations/#create-a-new-authorization) using basic authentication (without using the web flow) to create a token. I am not wanting to provide a client id and secret (read: can't provide). The closest thing I could find was [AuthorizationsClient.Create()](https://github.com/octokit/octokit.net/blob/master/Octokit/Clients/AuthorizationsClient.cs#L78) but it requires an id and string.

### PR DESCRIPTION
**Pull request by alfhenrik**
Implemented methods for creating an authorization without having to specify an app clientid/clientsecret.

These methods will ONLY work with username/password Basic Auth.

Added a helper method to return a GitHub client using basic auth credentials as if you have both password and oauth token environment variables, you get credentials based on the oauth token.

Fixes #969

### CODE REVIEW

File: Octokit.Reactive/Clients/ObservableAuthorizationsClient.cs [hunk 1]
```diff
@@ -46,6 +46,57 @@ public IObservable<Authorization> Get(int id)
...
+        /// Thrown when the current account has two-factor authentication enabled and an authentication code is required.
+        /// </exception>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        /// <returns>The created <see cref="Authorization"/>.</returns>
+        public IObservable<ApplicationAuthorization> Create(NewAuthorization newAuthorization)
+        {
+            Ensure.ArgumentNotNull(newAuthorization, "authorization");
+
+            return _client.Create(newAuthorization).ToObservable();
+
```
Comment (Haacked) [inline Octokit.Reactive/Clients/ObservableAuthorizationsClient.cs]: This PR is perfect so I'm going to nitpick on the extraneous line endings between the `return` statement and closing brace. :stuck_out_tongue:

File: Octokit/Clients/AuthorizationsClient.cs [hunk 2]
```diff
@@ -56,6 +56,77 @@ public Task<Authorization> Get(int id)
...
+                scopes = newAuthorization.Scopes,
+                note = newAuthorization.Note,
+                note_url = newAuthorization.NoteUrl,
+                fingerprint = newAuthorization.Fingerprint
+            };
+
+            var endpoint = ApiUrls.Authorizations();
+
+            return ApiConnection.Post<ApplicationAuthorization>(endpoint, requestData);
+
```
Comment (Haacked) [inline Octokit/Clients/AuthorizationsClient.cs]: Remove the extra line break.

**Comment by Haacked** [comment]
Nicely done! The only issue I had is with the white space formatting.

**Comment by alfhenrik** [comment]
D'Oh, white space!

**Comment by alfhenrik** [comment]
Whitespace gone the way of the dodo 😊

**Comment by shiftkey** [comment]
Thanks @alfhenrik!

**Comment by alfhenrik** [comment]
To be back submitting OSS PRs again!

Thank you guys!

**Comment by Haacked** [comment]
> Thank you guys!

No! THANK YOU HENRIK! :smile:

**Pull request merged**